llm/ggml.go

```diff
@@ -527,3 +528,22 @@ func (llm GGML) GraphSize(context, batch uint64) (kv, partialOffload, fullOffloa
 
 	return
 }
+
+// SupportsKVCacheType checks if the requested cache type is supported
+func (ggml GGML) SupportsKVCacheType(cacheType string) bool {
```

Comment:
To confirm: are there any model specific limitations to what cache type is supported? Or is that simplified now that f32 is not considered.

Reply:
E.g. to embedding models support q4_0? Ok if so, just confirming

Reply:
I believe technically you can load an embedding model with k/v cache quantisation but folks tend not to as it might impact the quality of the embeddings (hence why they're usually either f16 or q8_0).

This function originally had the check for embeddings and flash attention that are now the responsibility of the caller and is only called if SupportsFlashAttention is true - which is where the check for embeddings is done.